consensus/istanbul/validator/validator.go

```diff
@@ -96,7 +96,7 @@ func SelectRandomCommittee(validators []istanbul.Validator, committeeSize uint64
 
 	// ensure committeeSize and proposer indexes are valid
```

Comment:
I will take a look more. But I think if we pass filtered validator list to this function, maybe the logic could be simpler.